Sources/Asteroids/main.swift

```diff
@@ -0,0 +1,338 @@
+import FirebladeECS
+import Library
+import SDL2
+
+// MARK: SDL and Nexus Engines Setup
+
+// swiftlint:disable prefixed_toplevel_constant
+
+// initialize the SDL library with video and audio
+if SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO) != 0 {
+    fatalError("could not init video/audio")
+}
+
+// description of a display mode
+var displayMode = SDL_DisplayMode()
+// write info about current display into description
+SDL_GetCurrentDisplayMode(0, &displayMode)
+
+// timer for game loop
+var tFrame = Timer()
+// timer for app life time
+var tSetup = Timer()
+// number of frames passed during game
+var frameCount: UInt = 0
+// keeps track of frames per second
+var fps = 0.0
+
+// engine that orchestrates game entities and components
+let nexus = Nexus()
+
+// start tracking setup time
+tSetup.start()
+
+var windowTitle: String { "Asteroids ECS demo: [entities:\(nexus.numEntities) components:\(nexus.numComponents) families:\(nexus.numFamilies)] @ [FPS: \(fps), frames: \(frameCount)]" }
+
+// width and height for current display
+var width: Int32 = max(displayMode.w / 2, 800)
+var height: Int32 = max(displayMode.h / 2, 600)
+
+// flags for window to be created with
+let winFlags: UInt32 = SDL_WINDOW_SHOWN.rawValue // make window visible
+    | SDL_WINDOW_RESIZABLE.rawValue // and resizable
+// create window
+let hWin = SDL_CreateWindow(
+    windowTitle,
+    100, // x
+    100, // y
+    width,
+    height,
+    winFlags
+)
+
+// prevent game from running in case window can't be created
+if hWin == nil {
+    SDL_Quit()
+    fatalError("could not create window")
+}
+
+// provide game controls description
+func printHelp() {
+    let help: String = """
+    ================ FIREBLADE ASTEROIDS DEMO ===============
+    press:
+    ←          rotate spaceship to the left
```

Comment:
Some small improvement here would be if you would allow `W`,`A`, `S`, `D` for controlling the ship as well.